source/Irrlicht/CXMeshFileLoader.cpp

```diff
@@ -437,6 +438,8 @@ bool CXMeshFileLoader::readFileIntoMemory(io::IReadFile* file)
 		return false;
 	}
 	BinaryNumCount=0;
+	if (BinaryFormat)
+		End--; // remove null-terminator again, only needed for string-format

```

Comment:
`End` looks exclusive to me. So `End = Buffer + size` seems reasonable for binary format.

Reply:
is a mistake, will fix